src/main/java/Storage/Dbops.java

```diff
@@ -0,0 +1,128 @@
+            } else {
+                System.err.println("Memory file already exists.");
+            }
+        } catch (IOException e) {
+            printErrorOutput("An error occurred while creating memory file: " + e.getMessage());
+            exit(1);
+        }
+    }
+
+    /**
+     * Deletes the memory file, based on the filepath and filename.
+     */
+    private static void deleteMemoryFile() {
+        File fileToDelete = new File(Dbops.FILEPATH + Dbops.FILENAME);
+
+        if (fileToDelete.delete()) {
+            System.out.println("Old memory file deleted successfully.");
+        } else {
+            System.err.println("Failed to delete the old memory file.");
+        }
+    }
+
+    /**
+     * Loads tasks from the memory file.
+     *
+     * @param taskArray The TaskArray to load the tasks into.
+     */
+    public static void loadTasksFromFile(TaskArray taskArray) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(Dbops.FILEPATH + Dbops.FILENAME))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                // Create Task from each line and add them to the list
+                String[] parts = line.split(",");
+                boolean isMarked = Objects.equals(parts[2], "true");
+
+                if (Objects.equals(parts[0], "Todo") && parts.length == 3) {
+                    taskArray.add(new ToDo(parts[1], isMarked));
+                } else if (Objects.equals(parts[0], "Deadline") && parts.length == 4) {
+                    taskArray.add(new Deadline(parts[1], isMarked, parseDateTimeFromString(parts[3])));
+                } else if (Objects.equals(parts[0], "Event") && parts.length == 5) {
+                    taskArray.add(new Event(parts[1], isMarked, parseDateTimeFromString(parts[3]), parseDateTimeFromString(parts[4])));
+                } else {
+                    throw new DukeCorruptedDataException("Error: file is corrupted. Failed to load data from file.");
+                }
+            }
+        } catch (IOException e) {
+            String output = "File not found: " + Dbops.FILENAME + "\n" +
+                    "If this is your first day, welcome!\n" +
+                    "A new memory file, " + Dbops.FILENAME + " has been created.";
+
+            printBasicOutput(output);
+            Dbops.createMemoryFile();
+
+        } catch (DukeCorruptedDataException | DukeInvalidDateTimeException e) {
+            printErrorOutput(e + "\n");
+            Dbops.deleteMemoryFile();
+            Dbops.createMemoryFile();
+            printBasicOutput("A new memory file, " + Dbops.FILENAME + " has been created.");
+
+        }
+    }
+    /**
+     * Saves all tasks to the memory file.
+     *
+     * @param taskArray The TaskArray to load the tasks from.
+     */
+    public static void saveTasksToFile(TaskArray taskArray) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(Dbops.FILEPATH + Dbops.FILENAME))) {
```

Comment:
Wrong `try` implementation syntax, making the `try` statement look like a `while` loop lmao

You probably meant:
```java
try {
    BufferedWriter writer = new BufferedWriter(new FileWriter(Dbops.FILEPATH + Dbops.FILENAME));
    
    for (Task task : taskArray) {
        // code
    }
} catch (IOException e) {
    // code
}
```